### PR TITLE
Improve integration runtime cmdlets in data factory

### PR DESCRIPTION
**Pull request by zhangyd2015**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by zhangyd2015** [comment]
The build failed because the private build Microsoft.Azure.Management.DataFactory.0.4.0-preview is not signed. So this PR is still waiting for the official Microsoft.Azure.Management.DataFactory nuget.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/SetAzureDataFactoryIntegrationRuntimeCommand.cs [hunk 1]
```diff
@@ -163,7 +169,13 @@ public override void ExecuteCmdlet()
                     }
                     else
                     {
-                        resource.Properties = new SelfHostedIntegrationRuntime();
+                        var selfHosted = new SelfHostedIntegrationRuntime();
+                        if (AuthKey != null)
+                        {
+                            selfHosted.LinkedInfo = new LinkedIntegrationRuntimeKey(new SecureString(AuthKey));
```
Comment (cormacpayne) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/SetAzureDataFactoryIntegrationRuntimeCommand.cs]: @zhangyd2015 any reason that we can't just make the `AuthKey` parameter of type `SecureString`?

Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2/IntegrationRuntimes/SetAzureDataFactoryIntegrationRuntimeCommand.cs]: Change the type of AuthKey to SecureString.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 [hunk 2]
```diff
@@ -73,11 +85,6 @@ function Test-SelfHosted-IntegrationRuntime
 .SYNOPSIS
 Creates a SSIS-Azure integration runtime and then does operations.
 Deletes the created integration runtime at the end.
-
-Before running this test, make sure setting the following environment variables:
```
Comment (cormacpayne) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: @zhangyd2015 any reason for removing this comment?

File: src/ResourceManager/DataFactories/ChangeLog.md [hunk 3]
```diff
@@ -18,7 +18,8 @@
         - Additional information about change #1
 -->
 ## Current Release
-* Fix credential encryption issue that caused no meaningful error for some encryption operations.
+* Fix credential encryption issue that caused no meaningful error for some encryption operations
+* Enable integration runtime to be shared across data factory
```
Comment (cormacpayne) [inline src/ResourceManager/DataFactories/ChangeLog.md]: @zhangyd2015 your change should go in the DataFactoryV2 change log, not DataFactories

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/help/Set-AzureRmDataFactoryV2IntegrationRuntime.md [hunk 4]
```diff
@@ -61,13 +62,28 @@ The cmdlet updates the description of integration runtime named 'test-selfhost-i
 
 ## PARAMETERS
 
+### -AuthKey
+The authentication key of the self-hosted integration runtime.
+
+```yaml
+Type: String
```
Comment (cormacpayne) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/help/Set-AzureRmDataFactoryV2IntegrationRuntime.md]: @zhangyd2015 please regenerate the help content for this cmdlet

**Comment by cormacpayne** [comment]
@zhangyd2015 adding the `Do Not Merge` label on this PR until we are able to publish the updated SDK to NuGet

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 (hunk 2 above)
Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: This comment is not valid anymore. We don't use environment variables passing credential.

File: src/ResourceManager/DataFactories/ChangeLog.md (hunk 3 above)
Comment (zhangyd2015) [reply src/ResourceManager/DataFactories/ChangeLog.md]: Updated the change log in DataFactoryV2

**Comment by cormacpayne** [comment]
@zhangyd2015 changes look good to me -- waiting for this PR to reference the NuGet package instead of the signed package before merging

**Comment by zhangyd2015** [comment]
@cormacpayne I've updated Microsoft.Azure.Management.DataFactory nuget to published version and removed the one in localfeed.

**Comment by cormacpayne** [comment]
@zhangyd2015 there are merge conflicts since we merged in a separate PR containing changes to DataFactory. Would you mind pulling in those changes and resolving the conflicts?

**Comment by zhangyd2015** [comment]
@cormacpayne The conflict has been resolved.

**Comment by hvermis** [comment]
@cormacpayne Is there anything else blocking this PR? I don't have permission to Yandong's fork, but can we work together to get his changes in for this release?

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/496/

**Pull request merged**